Functions/ImageProcessing/xASL_im_ASLSubtractionAveraging.m

```diff
@@ -0,0 +1,388 @@
+% 1. Admin & checks
+% 2. Apply motion correction if this is needed
+%              
+% Nomenclature:
+% ASL4D = original timeseries
+% PWI4D = subtracted timeseries -> for quantification purposes (also applies to control4D)
+% PWI3D = averaged per TE—PLD—LabDur combination -> for QC purposes  (also applies to control3D)
+% PWI   = single perfusion-weighted volume -> for registration purposes  (also applies to control)
+% 
+% % we use the same nomenclature for control: control4D, control3D, control, as these are the same control-label pairs but without subtraction
+%
+% The quantification submodule xASL_wrp_Quantify only uses PWI4D from here, and recalculates PWI3D and PWI for quantification, potentially improving
+% quantification. So the user can, after QC, potentially remove some volumes from PWI4D, but should be aware that the information in the JSON sidecar
+% (TE, PLD, LabDur) should be adapted accordingly.
+%
+% EXAMPLE used by xASL_wrp_RegisterASL: PWI = xASL_im_ASLSubtractionAveraging(x, ASL_im);
+% EXAMPLE used by xASL_wrp_ResampleASL: [PWI, PWI3D, PWI4D, x] = xASL_im_ASLSubtractionAveraging(x, ASL4D, PWI4D, PWI3D);
+% __________________________________
+% Copyright (C) 2015-2023 ExploreASL
+
+
+%% ========================================================================================
+%% 1. Admin & checks
+% Input image volumes should be correct
+% Creation of the PWI3D or PWI4D image volumes can be skipped if they are provided as input
+
+% By default, we create 4D stuff, unless they are provided as input
+bCreatePWI4D = true;
+bCreateControl4D = true;
+
+% The 3D stuff is not created by default, because we need PWI4D OR Control4D for this
+bCreatePWI3D = false;
+bCreateControl3D = false;
+
+% The PWI/Control are not created by default, because we need PWI3D OR Control3D for this
+bCreatePWI = false;
+bCreateControl = false;
+
+% We skip the computation of PWI4D if it is provided
+if nargin>=3 && ~isempty(PWI4D)
+    if ~isnumeric(PWI4D)
+        error('Illegal PWI4D, it should be numerical');
+    elseif ndims(PWI4D)>4 || ndims(PWI4D)<3 % In theory, PWI4D can be a 3D matrix when it contains a single deltaM
+        error('PWI4D has an incorrect number of dimensions');
+	else
+        bCreatePWI4D = false; % All is correct and we don't have to compute PWI4D
+        bCreatePWI3D = true; % if PWI4D is provided, we compute PWI3D
+    end
+else
+    PWI4D = []; % dummy output
+end
+
+% We skip the computation of Control4D if it is provided
+if nargin>=5 && ~isempty(Control4D) % same as above
+    if ~isnumeric(Control4D)
+        error('Illegal Control4D, it should be numerical');
+    elseif ndims(Control4D)>4 || ndims(Control4D)<3
+        error('Control4D has an incorrect number of dimensions');
+	else
+        bCreateControl4D = false; % All is correct and we don't have to compute Control4D
+        bCreateControl3D = true; % if Control4D is provided, we compute Control3D
+    end
+    Control4D = []; % dummy output  
```

Comment:
there should be `else` here above this line.

Reply:
Done in 7f6f183131142fd00e507b045fb1a4928004b8cd